modules/spacelift/worker-pool/iam.tf

```
@@ -4,7 +4,7 @@ module "iam_label" {
 
   attributes = var.iam_attributes
 
-  context = module.this.context
+  context = merge(module.this.context, var.iam_context)
```

```suggestion
  tenant = var.teams_tenant_name
  stage = var.teams_stage_name
  environment = var.teams_environment_name
  
  context = module.this.context
```

This is a more opinionated suggestion than the other. The other context variables for `account-map` or `spacelift/spaces` are changed by an individual variable, rather than a merged object. IMO this should be consistent so that it's easier to understand

Set the default for each to `null`. Null label will take the value from context if not defined.

---

However taking a step back, why do we need to change these context options?

---

Aha, yeah, I should have explained this. In the end, I was looking at errors like this one:
```
An error occurred (AccessDenied) when calling the AssumeRole operation: User: 
arn:aws:sts::123456789012:assumed-role/acme-core-gbl-auto-spacelift-worker-pool/i-12345678901234567

is not authorized to perform: sts:AssumeRole on resource: 
arn:aws:iam::123456789013:role/acme-core-gbl-root-tfstate
```

---

When update the allowed roles in `tfstate-backend`, they are created with `gbl` as their environment. Given the choice between the two, it felt easier and less disruptive to allow the iam-role made in `spacelift/worker-pool` to change than to update `tfstate-backend` to allow roles that have something other than `gbl` in their role name.